src/stats/rawsender.cpp

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2017-2023 Vincent Thiery
+// Copyright (c) 2024 The Dash Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <stats/rawsender.h>
+
+#include <netaddress.h>
+#include <netbase.h>
+#include <util/sock.h>
+
+RawSender::RawSender(const std::string& host, uint16_t port, std::optional<std::string>& error) noexcept :
+    m_host{host},
+    m_port{port}
+{
+    if (host.empty()) {
+        error = "No host specified";
+        return;
+    }
+
+    if (auto netaddr = LookupHost(m_host, /*fAllowLookup=*/true); netaddr.has_value()) {
+        if (!netaddr->IsIPv4()) {
+            error = strprintf("Host %s on unsupported network", m_host);
+            return;
```

Comment:
Would like to avoid exceptions where possible, `StatsdClient` already prints errors generated and reaps the inactive object immediately ([source](https://github.com/dashpay/dash/blob/aad67b0839760494144afbf32d4fea1f4279b214/src/stats/client.cpp#L109-L113)), think it's fine as-is.